### ISSUE TITLE
Port kernel module of QtNetwork

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/qtbase-os2#37]
This is part of #35.

Actually, `QHostInfo` and `QNetworkInterfaceManager` are already done within #35. The remaining one is `QDnsLookup`. Not very easy, see https://github.com/bitwiseworks/qtbase-os2/issues/35#issuecomment-415900577.

**Comment by dmik** [comment]
Due to the absence of the new resolv API for OS/2, I decided to use a mutex and the old, non-thread-safe API in our `QDnsLookup` implementation.

Note that `QDnsLookup` class uses a thread pool to serve DNS query requests — so that a few lengthy requests could be executed in parallel. However, due to the mutex, this functionality is not available on OS/2: only one thread will be allowed to do a DNS query via `QDnsLookup` at a time and all other threads from the pool will have to wait until it finishes. This is not the best way to go, but still much better than providing a dummy implementation (like Android) which simply always fails with `NotSupported`.

More over, I don't have an impression this class is used very much. If and once we find a real world application that uses it and it starts annoying us, we will think of a better solution.
